tests/integration/test_data_cli.py

```diff
@@ -2,19 +2,86 @@
 
 from click.testing import CliRunner
 import pytest
-
+import json
+import httpx
+import respx
+from http import HTTPStatus
 from planet.cli import cli
 
+TEST_URL = 'https://api.planet.com/data/v1'
+TEST_QUICKSEARCH_URL = f'{TEST_URL}/quick-search'
+
+
+@pytest.fixture
+def invoke():
+
+    def _invoke(extra_args, runner=None):
+        runner = runner or CliRunner()
+        args = ['data'] + extra_args
+        return runner.invoke(cli.main, args=args)
 
-def test_data_command_registered():
+    return _invoke
+
+
+def test_data_command_registered(invoke):
     """planet-data command prints help and usage message."""
-    result = CliRunner().invoke(cli.main, ["data", "--help"])
+    runner = CliRunner()
+    result = invoke(["--help"], runner = runner)
     assert result.exit_code == 0
     assert "Usage" in result.output
     assert "search-quick" in result.output
     # Add other sub-commands here.
 
 
+@respx.mock
+@pytest.mark.asyncio
+@pytest.mark.parametrize("filter", [{''}, {"foo"}])
+@pytest.mark.parametrize(
+    "item_types", ['PSScene', 'SkySatScene', ('PSScene', 'SkySatScene')])
+def test_data_search_quick_filter_fail(invoke, item_types, filter):
+    """Test for planet data search_quick. Test with multiple item_types.
+    Test should fail as filter does not contain valid JSON."""
+    mock_resp = httpx.Response(HTTPStatus.OK,
+                               json={'features': [{
+                                   "key": "value"
+                               }]})
+    respx.post(TEST_QUICKSEARCH_URL).return_value = mock_resp
+
+    runner = CliRunner()
+    item_types = 'SkySatScene'
+    with pytest.raises(TypeError):
+        result = invoke(["search-quick", item_types, json.dumps(filter)],
+                        runner=runner)
```

Comment:
And here we no longer need json.dumps. `filter` is already (bad) JSON.

```suggestion
        result = invoke(["search-quick", item_types, filter], runner=runner)
```